start_control_ssm.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+log() {
+  echo "$*" >&2
+}
+
+enable_hybrid_env_ssm() {
+  # SSM parameters necessary to register hybrid environment Linux
+  failed_ssm_params=0
+  activation_code=$(get_user_data_ssm "activation-code")
+  activation_id=$(get_user_data_ssm "activation-id")
+  region=$(get_user_data_ssm "region")
+
+  # If any of the parameters are missing, then exit 1.
+  if [[ "${failed_ssm_params}" -ne 0 ]]; then
+    log "Missing required parameters to register with AWS Systems Manager (SSM)"
+    exit 1
+  fi
+
+  # Register with AWS Systems Manager (SSM)
+  amazon-ssm-agent -register -code "${activation_code}" -id "${activation_id}" -region "${region}"
+  if [[ "$?" -eq 1 ]]; then
+    log "Failed to register with AWS Systems Manager (SSM)"
+    exit 1
+  fi
+}
+
+# Extract the values from user-data json and count the failures (if any)
+get_user_data_ssm() {
+  local param_value
+  local ssm_param="${1:?}"
+  if ! param_value=$(jq --arg ssm_param "${ssm_param}" -e -r '.["ssm"][$ssm_param]' "${USER_DATA}"); then
+    log "Failed to parse ${ssm_param} from ${USER_DATA}"
+    ((++failed_ssm_params))
```

Comment:
Oh I see, sneaky.

Reply:
Yeah, too sneaky - can we return nonzero instead, and handle the counting in the other function?

Reply:
Rather than count, we can just exit 1. The counting was more of a hold over from the original design where we weren't going to exit if SSM failed to register.